### PR TITLE
[com_modules] - super user exception

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue  #18542.

### Summary of Changes
check only if not core.admin 

### Testing Instructions
see #18542

### CODE REVIEW

File: administrator/components/com_modules/models/modules.php [hunk 1]
```diff
@@ -314,9 +314,12 @@ protected function getListQuery()
 
 		// Filter by current user access level.
 		// Get the current user for authorisation checks
-		$user   = JFactory::getUser();
-		$groups = implode(',', $user->getAuthorisedViewLevels());
-		$query->where('a.access IN (' . $groups . ')');
+		if ($user->authorise('core.admin') !== true) 
+		{ 
+			$user   = JFactory::getUser();
```
Comment (csthomas) [inline administrator/components/com_modules/models/modules.php]: Take a look again:)

Comment (alikon) [reply administrator/components/com_modules/models/modules.php]: train approaching final destination
so
don't code
:+1:

File: administrator/components/com_modules/models/modules.php [hunk 2]
```diff
@@ -313,10 +313,10 @@ protected function getListQuery()
 		$query->where($db->quoteName('a.client_id') . ' = ' . (int) $clientId . ' AND ' . $db->quoteName('e.client_id') . ' = ' . (int) $clientId);
 
 		// Filter by current user access level.
+		$user   = JFactory::getUser();
```
Comment (Quy) [inline administrator/components/com_modules/models/modules.php]: Remove spaces before =.

File: administrator/components/com_modules/models/modules.php [hunk 3]
```diff
@@ -313,10 +313,13 @@ protected function getListQuery()
 		$query->where($db->quoteName('a.client_id') . ' = ' . (int) $clientId . ' AND ' . $db->quoteName('e.client_id') . ' = ' . (int) $clientId);
 
 		// Filter by current user access level.
-		// Get the current user for authorisation checks
 		$user   = JFactory::getUser();
-		$groups = implode(',', $user->getAuthorisedViewLevels());
-		$query->where('a.access IN (' . $groups . ')');
+		// Get the current user for authorisation checks
+		if ($user->authorise('core.admin') !== true) 
+		{
```
Comment (infograf768) [inline administrator/components/com_modules/models/modules.php]: Extraneous space after {

File: administrator/components/com_modules/models/modules.php [hunk 4]
```diff
@@ -313,10 +313,13 @@ protected function getListQuery()
...
-		// Get the current user for authorisation checks
 		$user   = JFactory::getUser();
-		$groups = implode(',', $user->getAuthorisedViewLevels());
-		$query->where('a.access IN (' . $groups . ')');
+		// Get the current user for authorisation checks
+		if ($user->authorise('core.admin') !== true) 
+		{ 
+			$groups = implode(',', $user->getAuthorisedViewLevels());
+			$query->where('a.access IN (' . $groups . ')');
+		}
```
Comment (infograf768) [inline administrator/components/com_modules/models/modules.php]: Delete indent

**Review by infograf768 — COMMENTED** [review]
cs

File: administrator/components/com_modules/models/modules.php (hunk 2 above)
Comment (alikon) [reply administrator/components/com_modules/models/modules.php]: removed

File: administrator/components/com_modules/models/modules.php (hunk 3 above)
Comment (alikon) [reply administrator/components/com_modules/models/modules.php]: removed

**Comment by infograf768** [comment]
Still missing deleting the tabs line 323 ;)

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on 418e91efa015ff07af4ba29f680e8035339dc7c3<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18548">issues.joomla.org/tracker/joomla-cms/18548</a>.</sub>

**Comment by csthomas** [comment]
Please make dron happy:
```
FILE: ...omla/joomla-cms/administrator/components/com_modules/models/modules.php
--------------------------------------------------------------------------------
FOUND 1 ERROR(S) AFFECTING 1 LINE(S)
--------------------------------------------------------------------------------
 317 | ERROR | Please consider a blank line preceding your comment
--------------------------------------------------------------------------------
UPGRADE TO PHP_CODESNIFFER 2.0 TO FIX ERRORS AUTOMATICALLY
```

**Comment by infograf768** [comment]
Drone is too polite to be human... 😺

**Comment by alikon** [comment]
drone still unhappy  but now not my fault :smile:

**Comment by rdeutz** [comment]
I restarted the job so drone is now happy too :-)

**Comment by infograf768** [comment]
As the corrected drone misbehavior has no impact on tests, we just need another tester to get this RTC

**Comment by ghost** [comment]
As i can't find Test Instructions on #18542 can @infograf768 you please give them?

**Comment by alikon** [comment]
Steps to reproduce the issue:

    1. Go into the backend of Joomla as super user and add a new module.
    2. Change module access to Guest Access.
    3. check that that module is not visbile in com_modules items list

Expected result

Module should be visible in module list  even if module access is set to Guset

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 1ad1308157ae95ad157603deb7d78505a6b8a00e<br /><br />Thanks for Test Instructions @alikon <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18548">issues.joomla.org/tracker/joomla-cms/18548</a>.</sub>

**Comment by ghost** [comment]
RTC after two successful tests.

**Comment by ladyjer** [comment]
I have tested this item :white_check_mark: successfully on 1ad1308157ae95ad157603deb7d78505a6b8a00e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18548">issues.joomla.org/tracker/joomla-cms/18548</a>.</sub>

**Pull request merged**